src/db/entry.rs

```diff
@@ -110,6 +116,60 @@ impl<'a> Entry {
     pub fn get_url(&'a self) -> Option<&'a str> {
         self.get("URL")
     }
+
+    /// Adds the current version of the entry to the entry's history
+    /// and updates the last modification timestamp.
+    /// The history will only be updated if the entry has
+    /// uncommited changes.
+    ///
+    /// Returns whether or not a new history entry was added.
+    pub fn update_history(&mut self) -> bool {
+        if self.history.is_none() {
+            self.history = Some(History::default());
```

Comment:
Since we have entry tags also being children of the history structs as well, if a user tries to edit a history entry, they would create a history which is valid in our library but KeePassXC and others will refuse to open the database.

Reply:
A possible solution might be to disallow write access to the history by making it private and only providing read methods. This goes against the rest of the API being as public as it is, which is probably another discussion to have before 1.0

Reply:
@sseemayer

> A possible solution might be to disallow write access to the history by making it private and only providing read methods

I agree with that. The `entries` could be private, and `add_entry` would be the only method that can modify the state of the history.

Reply:
Addressed in https://github.com/sseemayer/keepass-rs/pull/151